sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -121,8 +123,13 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
-    System.getProperty("java.io.tmpdir") + File.separator + db + ".db"
+  def getDatabasePath(dbName: String, path: Option[String]): String = {
+    val dbPath =
+      path.map(new Path(_))
+        .getOrElse(
+          new Path(new Path(System.getProperty("java.io.tmpdir")), dbName.toLowerCase() + ".db"))
+    val dbLocation = dbPath.toString
+    if (dbLocation.endsWith(File.separator)) dbLocation.dropRight(1) else dbLocation
```

Comment:
I guess `File.separator` is platform dependent, right? But, does the separator used by this platform is the separator used by the underlying FileSystem (the file system implemented based on Hadoop FS interface)? What is the intention of this line?